chia/wallet/sign_coin_spends.py

```diff
@@ -49,10 +52,17 @@ async def sign_coin_spends(
                 secret_key = await secret_key_for_public_key_f(pk)
             else:
                 secret_key = secret_key_for_public_key_f(pk)
-            if secret_key is None:
-                e_msg = f"no secret key for {pk}"
-                raise ValueError(e_msg)
-            assert bytes(secret_key.get_g1()) == bytes(pk)
+            if secret_key is None or secret_key.get_g1() != pk:
+                for derive in potential_derivation_functions:
+                    if inspect.iscoroutinefunction(secret_key_for_puzzle_hash):
+                        secret_key = await secret_key_for_puzzle_hash(derive(pk))
+                    else:
+                        secret_key = secret_key_for_puzzle_hash(derive(pk))
+                    if secret_key is not None and secret_key.get_g1() == pk:
+                        break
+                else:
+                    e_msg = f"no secret key for {pk}"
+                    raise ValueError(e_msg)
```

Comment:
since we don't use e_msg for anything else might as well put that string directly into ValueError